### PR TITLE
[Testing Audit] Animators

### PR DESCRIPTION
**Pull request by jtlan**
Here's the new test battery:
screen shot 2015-11-19 at 5 18 30 pm

Changes:
* `Animators.Null` actually has tests now
* `Animators.Easing` had its tests broken out to test the effects of step count, `startDelay()`, `stepDelay()`, and `stepDuration()` independently.

Should finish off #2622.

### CODE REVIEW

File: test/animators/easingAnimatorTests.ts [hunk 1]
```diff
@@ -2,153 +2,124 @@
...
-        //   |  |    |    |    |    |
-        //     0.0  0.5  1.0  1.5  2.0
-        let actualIterativeDelay = (<any>animator)._getAdjustedIterativeDelay(iterationSteps);
-        assert.strictEqual(actualIterativeDelay, expectedIterativeDelay,
-          "The total duration constraint is just an upper bound");
+      it("restricts the total time when the step delay increases", () => {
+        animator.stepDelay(2 * ORIGINAL_STEP_DELAY);
+        const longerStepDelayTotalTime = animator.totalTime(NUM_STEPS);
+        assert.closeTo(longerStepDelayTotalTime, totalTimeLimit, EPSILON,
+          "adding more steps does not increase the total time past the limit");
```
Comment (ztsai) [inline test/animators/easingAnimatorTests.ts]: increasing step delay...

File: test/animators/easingAnimatorTests.ts [hunk 2]
```diff
@@ -2,153 +2,124 @@
...
       });
 
+      it("restricts the total time when the step duration increases", () => {
+        animator.stepDuration(2 * ORIGINAL_STEP_DURATION);
+        const longerStepDurationTotalTime = animator.totalTime(NUM_STEPS);
+        assert.closeTo(longerStepDurationTotalTime, totalTimeLimit, EPSILON,
+          "adding more steps does not increase the total time past the limit");
```
Comment (ztsai) [inline test/animators/easingAnimatorTests.ts]: increasing step duration

**Comment by ztsai** [comment]
can you also test that `Easing.animate` applies changes to attribute?

**Comment by jtlan** [comment]
>can you also test that Easing.animate applies changes to attribute?

I tried that, but because it uses an asynchronous transition it doesn't finish before the test code goes to check that the attributes were applied.

**Pull request merged**